src/ica_api.c

```diff
@@ -1669,10 +1676,14 @@ void ica_ec_key_free(ICA_EC_KEY *key)
 	free(key);
 }
 
-static inline int check_fips(void)
+static inline int check_fips(unsigned int nid)
```

Comment:
I would prefer a function name, which better describes, what this function is checking, e.g. `fips_restricted()` or similar. While at it, it might be moved to fips.c?